packages/smooth_app/lib/generic_lib/loading_dialog.dart

```diff
@@ -99,18 +99,10 @@ class LoadingDialog<T> {
       body: FutureBuilder<T>(
         future: future,
         builder: (BuildContext context, AsyncSnapshot<T> snapshot) {
-          if (snapshot.connectionState == ConnectionState.done) {
-            // Now it's either hasError or successful.
-            // We cannot check hasData because data can be null or void.
-            if (snapshot.hasError) {
-              return ListTile(
-                title: Text(appLocalizations.error_occurred),
-              );
-            }
+          if (snapshot.hasError || snapshot.hasData) {
```

Comment:
Technically the snapshot may have finished but have no data (if the normal result is `null`), therefore I would suggest to put back something like `snapshot.connectionState == ConnectionState.done`.